frontend/src/v5/ui/routes/dashboard/projects/federations/federationSettingsForm/federationSettingsForm.component.tsx

```diff
@@ -0,0 +1,405 @@
+/**
+ *  Copyright (C) 2022 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useEffect } from 'react';
+import { formatMessage } from '@/v5/services/intl';
+import { FormattedMessage } from 'react-intl';
+import { yupResolver } from '@hookform/resolvers/yup';
+import * as Yup from 'yup';
+import { TextField, MenuItem, InputLabel, FormControl } from '@material-ui/core';
+import { Select } from '@controls/select';
+import { FormModal } from '@/v5/ui/controls/modal/formModal/formDialog.component';
+import { useForm, SubmitHandler, Controller } from 'react-hook-form';
+import { useParams } from 'react-router';
+import { FederationsActionsDispatchers } from '@/v5/services/actionsDispatchers/federationsActions.dispatchers';
+import { IFederation, EMPTY_VIEW, RawFederationSettings } from '@/v5/store/federations/federations.types';
+import { ShareTextField } from '@controls/shareTextField';
+import { FlexContainer, SectionTitle, Thumbnail, ThumbnailPlaceholder, SelectView, ViewLabel, MenuItemView, UnitTextField } from './federationSettingsForm.styles';
+
+const UNITS = {
+	mm: 'Millimetres',
+	cm: 'Centimetres',
+	dm: 'Decimetres',
+	m: 'Metres',
+	ft: 'Feet and inches',
+};
+
+interface IFormInput {
+	name: string;
+	unit: string;
+	desc?: string;
+	code?: string;
+	defaultView: string;
+	latitude: number;
+	longitude: number;
+	angleFromNorth: number;
+	x: number;
+	y: number;
+	z: number;
+}
+
+const getDefaultValues = (federation: IFederation) => {
+	const { unit = 'mm', angleFromNorth = 0 } = federation.settings || {};
+	const {
+		latLong,
+		position,
+	} = federation.settings?.surveyPoint || {};
+	const [x, y, z] = position || [0, 0, 0];
+	const [latitude, longitude] = latLong || [0, 0];
+	const { code, name, description: desc } = federation;
+	const defaultView = federation?.settings?.defaultView || EMPTY_VIEW._id;
+	return {
+		name,
+		desc,
+		code,
+		unit,
+		defaultView,
+		latitude,
+		longitude,
+		angleFromNorth,
+		x,
+		y,
+		z,
```

Comment:
I think this is the data structure that the original settings should be transformed to when fetching the settings. That way, all this will be unnecessary. 

Then when updating the settings it would transform back to the format the backend understand.

Reply:
Imo, neither option is too clear. 
**Option 1:** if you try to access `settings.x` it is not very clear what `x` stands for;
**Option 2:** similarly, if you access `settings.surveyPoint.position[0]`, `0` is a magic number, and it is not very clear what you are getting access to.

In theory, I believe that the most readable structure is one that still contains a surveyPoint property, but which is structured as follow:
```js

settings: {
    surveyPoint: {
          latitute: number,
          longitude: number,
          x: number,
          y: number,
          z: number,
    }
    ...
}
``` 
If any, I would go with this new structure, but I am not sure whether this is a good idea as it drifts away a bit too much from the backend data. If so, I would with option 2 and I would either create constant denoting the `position` elements, just like this:

```js
const POSITION: {
  X: 0,
  Y: 1,
  Z: 2
}
```
so that we can access the surveyPoint as this:
```js
settings.surveyPoint.position[POSITION.X]
```
or rename the `position` property to `positionXYZ` so, just like for `latLong` you know immediately that index `0` will access `x`

Reply:
Imo. the position remapping is a bit overkill. We assume array of types everywhere (just like we never do rgba labelling on colours, we just use an array[4] or array[3].

Simiarly, as a graphics application where a label defines something to be obviously a point in 3d world, it's a bit overkill to do labelling like this.

If you really want to group the values into a embedded object then group it like the form 
![image](https://user-images.githubusercontent.com/11945337/152534927-e4fa6f3f-ffb9-4596-b0fa-e294e3a0b0ba.png)

i.e. angle from north should also be inside.

Reply:
So you would still keep the `surveryPoint` property, but also including the angleFromNorth, right? @carmenfan

Reply:
@Amantini1997 I don't have a strong feeling about the `surveyPoint` property, so up to you and @sanmont3drepo